dotnet/samples/GettingStarted/Agents/Agent_Step20_BackgroundResponsesWithToolsAndPersistence/Program.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+// This sample demonstrates how to use background responses with ChatClientAgent and Azure OpenAI Responses for long-running operations.
+// It shows polling for completion using continuation tokens, function calling during background operations,
+// and persisting/restoring agent state between polling cycles.
+
+#pragma warning disable CA1050 // Declare types in namespaces
+
+using Azure.AI.OpenAI;
+using Azure.Identity;
+using Microsoft.Agents.AI;
+using Microsoft.Extensions.AI;
+using OpenAI;
+
+var endpoint = Environment.GetEnvironmentVariable("AZURE_OPENAI_ENDPOINT") ?? throw new InvalidOperationException("AZURE_OPENAI_ENDPOINT is not set.");
+var deploymentName = Environment.GetEnvironmentVariable("AZURE_OPENAI_DEPLOYMENT_NAME") ?? "gpt-5";
+
+var stateStore = new AgentStateStore();
```

Comment:
Suggest bringing the `AgentStateStore` impl, with PersistAgentStore being a LocalFunction as part of the `Program.cs`.

```suggestion
```

Reply:
IMHO, this counter intuitively also makes the sample harder to read (as now the developer needs to jump around other files to get the full understanding. Myself when I was reading the PR for a moment though there was a `AgentStateStore` component in the package. 

We moved away from this approach in benefit of end-to-end samples, that can potentially be used as a single file sample in .NET 10.